knative/deployer.go

```diff
@@ -90,7 +91,20 @@ func (d *Deployer) Deploy(f faas.Function) (err error) {
 	return nil
 }
 
-func generateNewService(name, image string) *servingv1.Service {
+func probeFor(port, url string) *corev1.Probe {
+	return &corev1.Probe{
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: url,
+				Port: intstr.IntOrString{
```

Comment:
`intstr.FromString(port)` might be better.
But leads to validation error mentioned before.

Reply:
You should be able to drop this completely. The "correct" port is inferred. The readiness endpoint has to be served on the "default" server though (the one listening on `$PORT`).